airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/CsvRecordBuffer.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.s3.csv;
+
+import io.airbyte.commons.functional.CheckedBiFunction;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.destination.record_buffer.BaseRecordBufferImplementation;
+import io.airbyte.integrations.destination.record_buffer.FileRecordBuffer;
+import io.airbyte.integrations.destination.record_buffer.RecordBufferImplementation;
+import io.airbyte.integrations.destination.record_buffer.RecordBufferStorage;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.lang3.StringUtils;
+
+public class CsvRecordBuffer extends BaseRecordBufferImplementation {
+
+  private CSVPrinter csvPrinter;
+  private final CsvSheetGenerator csvSheetGenerator;
+
+  protected CsvRecordBuffer(final RecordBufferStorage bufferStorage, final CsvSheetGenerator csvSheetGenerator) throws Exception {
+    // we always want to compress csv files
+    super(bufferStorage, true);
+    this.csvSheetGenerator = csvSheetGenerator;
+  }
+
+  @Override
+  protected void createWriter(final OutputStream outputStream) throws IOException {
+    csvPrinter = new CSVPrinter(new PrintWriter(outputStream, true, StandardCharsets.UTF_8), CSVFormat.DEFAULT);
```

Comment:
are we always using the same `CsvFormat`? Looks like we're using `QuoteMode.ALL` for `GcsCsvWriter` and `AzureBlobStorageCsvWriter`

Reply:
I added a `withCsvFormat` method to configure the writer options